search/src/main/scala/ch.epfl.scala.index.search/DataRepository.scala

```diff
@@ -1,41 +1,53 @@
 package ch.epfl.scala.index.search
 
-import java.io.File
-
 import ch.epfl.scala.index.model._
 import ch.epfl.scala.index.model.misc.{Pagination, _}
 import ch.epfl.scala.index.model.release._
 import ch.epfl.scala.index.search.mapping._
-import com.sksamuel.elastic4s.bulk.RichBulkResponse
-import com.sksamuel.elastic4s.embedded.LocalNode
-import com.sksamuel.elastic4s.searches.queries.QueryDefinition
-import com.sksamuel.elastic4s.searches.sort.SortDefinition
-import com.sksamuel.elastic4s.{
-  ElasticDsl,
-  ElasticsearchClientUri,
-  HitReader,
-  TcpClient
-}
 import com.typesafe.config.ConfigFactory
 import com.typesafe.scalalogging.LazyLogging
-import org.elasticsearch.action.bulk.BulkResponse
-import org.elasticsearch.cluster.health.ClusterHealthStatus
-import org.elasticsearch.common.lucene.search.function.CombineFunction
-import org.elasticsearch.common.lucene.search.function.FieldValueFactorFunction.Modifier
-import org.elasticsearch.search.sort.SortOrder
+import org.testcontainers.containers.BindMode
+import org.testcontainers.elasticsearch.ElasticsearchContainer
+import org.testcontainers.utility.DockerImageName
 
-import scala.jdk.CollectionConverters._
+import java.io.{Closeable, File}
+import java.nio.file.Files
+import java.nio.file.attribute.PosixFilePermission
 import scala.concurrent.{ExecutionContext, Future}
-import java.io.Closeable
+import scala.jdk.CollectionConverters._
+import com.sksamuel.elastic4s.{ElasticClient, ElasticDsl}
+import com.sksamuel.elastic4s.requests.common.HealthStatus
+import com.sksamuel.elastic4s.requests.bulk.BulkResponse
+import com.sksamuel.elastic4s.requests.bulk.BulkResponseItem
+import com.sksamuel.elastic4s.HitReader
+import com.sksamuel.elastic4s.requests.searches.queries.Query
+import com.sksamuel.elastic4s.requests.searches.sort.SortOrder
+import com.sksamuel.elastic4s.ElasticProperties
+import com.sksamuel.elastic4s.requests.searches.queries.funcscorer.FieldValueFactorFunctionModifier
+import com.sksamuel.elastic4s.requests.searches.queries.funcscorer.CombineFunction
+import com.sksamuel.elastic4s.requests.searches.sort.Sort
+import com.sksamuel.elastic4s.analysis.Analysis
+import com.sksamuel.elastic4s.requests.mappings.MappingDefinition
+import com.sksamuel.elastic4s.requests.searches.aggs.responses.bucket.Terms
+import com.sksamuel.elastic4s.http.JavaClient
 
 /**
  * @param esClient TCP client of the elasticsearch server
  */
-class DataRepository(esClient: TcpClient, indexName: String)(implicit
+class DataRepository(
+    esClient: ElasticClient,
+    container: Option[ElasticsearchContainer],
```

Comment:
I don't like having his here, I am not sure of how to manage it :/
It is here only for `close` if I understand correctly...

Reply:
I agree. A potential solution is to create a `Closeable` abstraction on top of `ElasticClient` that can encapsulate the container.
But the more I think of it, the more I convince myself that the container lifecycle should be managed by sbt. So I don't want to add more code that I will probably revert soon.

Reply:
The container is required only for testing. So tests could be another point where this could be handled instead of sbt

Reply:
It is also required for running the Scaladex locally. So it's rather a matter of local/remote than a matter of main/test.